### ISSUE TITLE
0.4.2 terrafrom refresh not working for aws security groups

### ISSUE THREAD

**Issue by vitalis** [hashicorp/terraform#1535]
In version 0.4.0 the refresh was working fine, after update to 0.4.2 it does not update the data.

Output from 0.4.0:
```json
"aws_security_group.application": {
    "type": "aws_security_group",
    "primary": {
        "id": "sg-7ffc4816",
        "attributes": {
            "description": "Instances running application servers",
            "egress.#": "0",
            "id": "sg-7ffc4816",
            "ingress.#": "2",
            "ingress.2214680975.cidr_blocks.#": "1",
            "ingress.2214680975.cidr_blocks.0": "0.0.0.0/0",
            "ingress.2214680975.from_port": "80",
            "ingress.2214680975.protocol": "tcp",
            "ingress.2214680975.security_groups.#": "0",
            "ingress.2214680975.self": "false",
            "ingress.2214680975.to_port": "80",
            "ingress.2617001939.cidr_blocks.#": "1",
            "ingress.2617001939.cidr_blocks.0": "0.0.0.0/0",
            "ingress.2617001939.from_port": "443",
            "ingress.2617001939.protocol": "tcp",
            "ingress.2617001939.security_groups.#": "0",
            "ingress.2617001939.self": "false",
            "ingress.2617001939.to_port": "443",
            "name": "application",
            "owner_id": "xxxxxxxxxxxx"
        }
    }
}
```

Output from 0.4.2:
```json
"aws_security_group.application": {
    "type": "aws_security_group",
    "primary": {
        "id": "sg-7ffc4816",
        "attributes": {
            "description": "Instances running application servers",
            "egress.#": "0",
            "id": "sg-7ffc4816",
            "name": "application",
            "owner_id": "xxxxxxxxxxxx"
        }
    }
}
```

**Comment by vitalis** [comment]
Update: version 0.4.1 is working fine

**Comment by catsby** [comment]
Thanks for reporting, we'll take a look

**Comment by catsby** [comment]
Do you have a sanitized config file that reproduces this, or any other steps to reproduce this? I've tried with 0.4.0 through 0.4.2 and get the same results, the full attributes are listed.

**Comment by vitalis** [comment]
I followed the suggestion of adding existing resources to to state.
I updates my `terraform.tfstate` with the following data:
```json
"aws_security_group.application": {
    "type": "aws_security_group",
    "primary": {
        "id": "sg-7ffc4816"
    }
}
```

then used `terrafrom refresh` command

**Comment by vitalis** [comment]
I also see another issue in 0.4.1 ingress configuration:
Here is the list of existing ingress on security group:
![ec2_management_console](https://cloud.githubusercontent.com/assets/36814/7178781/210586bc-e43a-11e4-9fa8-ad5d4e78406f.png)

and here is the result I have after `terrafrom refresh`:

```json
"aws_security_group.mongo": {
    "type": "aws_security_group",
    "primary": {
        "id": "sg-79191d10",
        "attributes": {
            "description": "MongoDB Servers",
            "egress.#": "0",
            "id": "sg-79191d10",
            "name": "mongo",
            "owner_id": "xxxxxxxxxxxxxxxxx"
        }
    }
}
```

**Comment by radeksimko** [comment]
> I followed the suggestion of adding existing resources to the state.

where did you get such suggestion?

There's an [existing feature request](https://github.com/hashicorp/terraform/issues/581) for this, but I don't think it's a good idea to manipulate with tfstate file manually.

**Comment by vitalis** [comment]
https://www.dark.ca/2015/01/27/handling-extant-resources-in-terraform/

**Comment by nevins-b** [comment]
I think #1597 addresses this issue.

**Comment by mitchellh** [comment]
Fixed by #1597
